murakami/runners/ooniprobe.py

```diff
@@ -0,0 +1,174 @@
+                )
+            except subprocess.CalledProcessError as e:
+                raise RunnerError(
+                    "ooniprobe run unattended returned a non-zero exit code. (err: " + output.stderr + ")"
+                )
+            endtime = datetime.datetime.utcnow()
+           
+            # If the previous commands succeeded, then we can parse the output
+            # of "ooniprobe list --batch", which only contains the results of
+            # the current run (since the database was emptied beforehand).
+            logging.info("Reading ooniprobe results...")
+            cmdargs = [
+                "ooniprobe",
+                "list",
+                "--batch"
+            ]
+
+            try:
+                output = subprocess.run(
+                    cmdargs,
+                    check=True,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE
+                )
+            except subprocess.CalledProcessError as e:
+                raise RunnerError(
+                    "ooniprobe list --batch returned a non-zero exit code. (err: " + output.stderr + ")"
+                )
+            
+            # Parse the output of "ooniprobe list --batch". The output is in
+            # JSONL format, and we need to only parse lines containing an
+            # "id" field.
+            results = []
+            try:
+                results = [json.loads(line) for line in output.stdout.splitlines()]
+            except json.decoder.JSONDecodeError as e:
+                raise RunnerError(
+                    "ooniprobe list --batch returned invalid JSON. (err: " + str(e) + ")"
+                )
+
+            test_results = {}
+            for js in results:
+                # If the JSON contains an "id" field, then it is a test result.
+                # Get the corresponding test data with:
+                # "ooniprobe list <id> --batch".
+                if "id" in js["fields"]:
+                    # Get the test's name.
+                    test_name = js["fields"]["name"]
+
+                    cmdargs = [
+                        "ooniprobe",
+                        "list",
+                        str(js["fields"]["id"]),
+                        "--batch"
+                    ]
+
+                    try:
+                        output = subprocess.run(
+                            cmdargs,
+                            check=True,
+                            stdout=subprocess.PIPE,
+                            stderr=subprocess.PIPE
+                        )
+                    except subprocess.CalledProcessError as e:
+                        raise RunnerError(
+                            "ooni probe list <id> --batch returned a non-zero exit code. (err: " + output.stderr + ")"
+                        )
+
+                    try:
+                        test_measurements = [json.loads(line) for line in output.stdout.splitlines()]
```

Comment:
This code is not 100% correct. You should only include JSONs with fields.type == "measurement_item".